src/extension.ts

```diff
@@ -48,191 +58,18 @@ export function activate(context: vscode.ExtensionContext) {
-			text-align: center;
-			padding: 2%;
-		}
-
-		.block{
-		  background-color: #00C896;
-		  width: 90%;
-		  min-height: 15%;
-		  margin: 1% auto;
-		  display: flex;
-		  justify-content: center;
-		  flex-direction: column;
-		  align-items: center;	
-		  transform: scaleY(-1);
-		}
-		  .center{
-			text-align: center;
-		  }
-		  ::-webkit-scrollbar{
-			width: 6px;
-		  }
-		  
-		  ::-webkit-scrollbar-thumb{
-			border-radius: 10px;
-			background-color: #eee;
-			height: 5px;
-		  }
-		  
-		  ::-webkit-scrollbar-thumb:window-inactive{
-			background: rgba(100,100,100,0.4);
-		  }
-		  
-		  .block-attrib:hover{
-			background-color: #06ebb2;
-			cursor: pointer;
-		  }
-
-		  .tree-node {
-			border: 1px solid #fff;
-			height: 200px;
-			width: 200px;
-			background-color: #242f38;
-			border-radius: 50%;
-			margin-bottom: 200px;
-			text-align: center;
-			vertical-align: middle;
-		  }
-		</style>
-			<meta charset="UTF-8">
-			<meta name="viewport" content="width=device-width, initial-scale=1.0">
-			<title>Stack Frame Visualization</title>
-		</head>
-		<body>
-		
-		<h2 align = "center">Stack Visualization (Stack View)</h2>
-		<div id="stack-container">
-			<div id="stack"></div>
-		</div>
-
-	
-		<h2 align = "center">Stack Visualization (Tree View)</h2>
-		<div id="trees-container">
-		</div>
-
-		<sf id="sf-id" data-frames='${sfs}' ></sf>
-		
-		<script>
-			const framesObj = document.querySelector('#sf-id');
-			var jsonFrames = JSON.parse(framesObj.dataset.frames);
-			
-			var stack =  document.getElementById("stack");
-			for (let frame of jsonFrames) 
-			{
-				var fullText = '<div class="block">';
-				var innerTxt = '<h1 class="title">' + frame.name.split('.').pop() + '</h1>';
-				
-				if(frame.variables !== undefined && frame.variables !== null)
-				{
-					for(let x of frame.variables)
-					{
-						innerTxt += '<h2 class="block-attrib"> ' + x.name + ':' + x.value + '</h2>';
-					}
-				}
-				fullText += innerTxt;
-				fullText += "</div>"
-				stack.innerHTML = fullText + stack.innerHTML;
-				stack.scrollTop = stack.scrollHeight;
-			}
-			console.log(fullText);
-			
-			/*
-			var treesContainer = document.getElementById("trees-container"); 
-			for (let frame of jsonFrames) {
-				var fullTreeStructure = '';
-				for (let frame of jsonFrames) 
-				{
-					var innerTreeStructure = '<div class="tree-node"><b>' + frame.name.split('.').pop() + '</b> <br>';
-
-					if(frame.variables !== undefined && frame.variables !== null)
-					{
-						for(let x of frame.variables)
-						{
-							innerTreeStructure += '<p style="color:green"> ' + '( ' + x.name + ':' + x.value + ' )+ '</p>';
-						}
-					}
-
-					innerTreeStructure += '</div>';
-
-					fullTreeStructure += innerTreeStructure;
-				}
-
-				treesContainer.innerHTML = fullTreeStructure;
-		
-			}
-			*/
-		</script>
-		</body>
-		</html>`;
-
-	
+		return fs.readFileSync(extensionPath + "/media/index.html").toString();
+	}
+	else{
+		//very inefficient 
```

Comment:
Could we refer to this https://code.visualstudio.com/api/extension-guides/webview#passing-messages-from-an-extension-to-a-webview to see if we can pass the variable to the webview this way ?

Reply:
Working on it